neuland/simulation/generator/generator.h

```diff
@@ -0,0 +1,137 @@
+#pragma once
+#include <Math/GenVector/Cartesian3D.h>
+#include <Math/GenVector/Polar3D.h>
+#include <Math/GenVector/PxPyPzE4D.h>
+#include <algorithm>
+#include <cmath>
+#include <fmt/format.h>
+#include <TRandom.h>
+
+
+template <typename ADist, typename EDist>
+class TrackGenerator
+{
+  public:
+    TrackGenerator(const ADist& AngleDist,const EDist& EnergyDist)
+        : AngleDist_{ AngleDist }
+        , EnergyDist_{ EnergyDist }
+    {
+    }
+    void set_detector_width(double width);
+    void set_detector_length(double length);
+    void set_detector_height(double height);
+
+  private:
+    double detector_width{};
+    double detector_length{};
+    double detector_height{};
+
+    double particle_mass{};
+    static constexpr double speed_of_light{ 299792458 };
```

Comment:
speed of light value has been defined in [R3BNeulandCommon.h](https://github.com/R3BRootGroup/R3BRoot/blob/dev/neuland/shared/R3BNeulandCommon.h).